PwaUriHandler/explainer.md

```diff
@@ -0,0 +1,250 @@
+
+Example web app manifest hosted at `www.contoso.com`:
+
+```json
+{
+  "name": "Contoso App",
+  "description": "A Business App",
+  "uri_handlers": [
+      {
+          "base": "www.contoso.com",
+          "paths": ["*"],
+          "excludePaths": [
+              "/about",
+              "/blog",
+              "/privacy"
+          ]
+      },
+      {
+          "base": "%*.contoso.com",
+          "paths": ["*"]
+      },
+      {
+          "base": "www.conto.so",
+      }
+  ],
+  "icons": [
+    {
+      "src": "images/icons-144.png",
+      "type": "image/png",
+      "sizes": "144x144"
+    }
+  ],
+  "background_color": "#FFFFFF",
+  "display": "standalone",
+  "start_url": "/",
+}
+```
+
+These are the fields in each URI handler object:
+
+| Field     | Required / Optional | Description                                      | Default                           |
+|:----------|:--------------------|:-------------------------------------------------|:----------------------------------|
+| `base`    | Required            | Base portion of the URI to be handled. Can also be a complete URI. | N/A             |
+| `paths`   | Optional            | Array of allowed paths relative to `base`        | `[]`                              |
+| `excludePaths` | Optional       | Array of disallowed paths relative to `base`     | `[]`                              |
+
+A URI matches a URI handler if it matches the `base`, at least one of values in `paths` if there are any, and does not match any of the values in `excludePaths`.
+
+Requested URIs do not have to be within the requesting PWA's scope. In this scheme, any URI can be registered as part of the URI handling request. The `base` field is necessary because URIs from different domains can be requested. Not restricting URIs to the same scope or domain as the requesting PWA gives the developer freedom to use multiple domain names for the same content and handle them with the same PWA. See [this section](#pwa-to-site-association) for how cross-domain requests are validated. The `base` field can start with a `%*.` prefix to indicate the inclusion of subdomains.
+
+(Implementation note: URI handling requests are registered with either the browser or the OS when a PWA is being installed. At this point, the browser should validate the requests. If necessary, the PWA install can be failed.)
+
+#### Wildcard Matching
+
+Wildcard characters can be used in the values of all three fields: `base`, `paths`, and `excludePaths`. The wildcard `*` matches zero or more characters. The wildcard `?` matches exactly one character.
+
+The `base` field is able to contain a wildcard prefix to allow the specification of sub-domains: eg. `%*.contoso.com` matches `jadams.contoso.com` and `www.jqadams.contoso.com` but not `contoso.com`.
+
+### PWA to site association
```

Comment:
Another proposed mechanism that aims to solve a similar attribution problem is [First Party Sets](https://github.com/krgovind/first-party-sets). We might want to acknowledge that proposal explicitly or acknowledge that there may be other means of solving for this use case.

Reply:
How about: 

"There may be other ways of specifying a group of sub-domains using proposals such as [First Part Sets](https://github.com/krgovind/first-party-sets)."

Reply:
Perhaps replace "sub-domains" with "related domains" and your updated language should be good!

Reply:
Edited.